bodhi/server/models.py

```diff
@@ -3339,7 +3365,7 @@ def met_testing_requirements(self):
         if min_num_days:
             if not self.meets_testing_requirements:
                 return False
-        else:
+        elif not self.autopush:
```

Comment:
Why this change?

Reply:
In case of rawhide min_num_days will be 0 which would make us think that we already met the testing requirements.

Reply:
If the requirement is 0, then we have met the requirements in that case. I think we should keep the concept of autopush and requirement separate. The push date isn't a requirement, it's just an automation.

Reply:
Ok that makes sense to me, but for some reason in the `approve_testing` script, we do this (see below) if we have met the requirements, which mean we will never push the rawhide updates. 

https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/scripts/approve_testing.py#L85

The script might be wrong or not, I am not sure :stuck_out_tongue_closed_eyes:

Reply:
@cverna ah yes, I am familiar with that and do not like it!

I think the point of it is to avoid writing a comment on updates that could be pushed every time the script runs, so it's trying to detect if the update has already been commented on. It's not a good way to do that, but I think that's what it's trying to do, if that makes sense.

I think it'd be better if there were:

0. A formal way to mark a Bodhi comment as being the comment that says this. Right now, it checks if the comment was authored by bodhi and has that particular text (ewww). It'd be better if the Comment model had some metadata/enum to mark certain Bodhi comments as being of that type - then we can more cleanly detect whether Bodhi has already added that comment to an update.
1. I don't love that we have ```met_testing_requirements()``` *and* ```meets_testing_requirements()```. The names different in tense, of course, but really I think we just want to make sure we don't comment repeatedly.

Anyways, we don't need to solve these nasty problems in this pull request - I'm just trying to give you some context for what this code seems like it's trying to do. It's confusing…

Reply:
Another hilarious thing about those two functions is that one of them calls the other!